### DISCUSSION TITLE
recommended mobile test framework

### DISCUSSION THREAD

**Discussion by pansfy** [seleniumbase/SeleniumBase#848] [Ideas]
hello, sir.
Can the framework be integrated with appium to integrate mobile testing for android or ios?
Please help recommend a similar mobile testing framework with many useful and easy to use?

**Comment by mdmintz** [comment]
Hi @pansfy , you can use SeleniumBase for mobile testing: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/mobile_testing.md
Real mobile web browser emulation.

**Reply by pansfy** [reply]
I understand the use of web-based simulation, but i am looking for solutions for testing native apps or hybrid apps.
Do you have any good suggestions, thank you.

**Reply by mdmintz** [reply]
Aside from Appium, which I haven't used, you might be able to use a mobile device farm from BrowserStack or Sauce Labs. I'm only familiar with mobile emulators for mobile testing.

**Reply by pansfy** [reply]
ok, thanks very much.

**Reply by narendarbhogala** [reply]
@pansfy , have you integrated Appium with SeleniumBase. Curious to know if that is possible?

**Reply by Jeremy-QA** [reply]
@narendarbhogala Yes, download all the necessary Appium dependencies and import them into your project. Use pytest to manage the interaction between Appium and SeleniumBase.
